lib/3scale/backend/async_redis/client.rb

```diff
@@ -31,32 +31,49 @@ def url_present?(url)
           def connect_tcp(opts)
             uri = URI(opts[:url])
 
-            credentials = [ uri.user || opts[:username], uri.password || opts[:password]]
-            db = uri.path[1..-1]
-
-            protocol = Protocol::ExtendedRESP2.new(db: db, credentials: credentials)
+            database = uri.path[1..-1]
+            credentials = [ uri.user || opts[:username], uri.password || opts[:password]].compact
+            credentials = nil unless credentials.any?
```

Comment:
Same 👼 

```suggestion
            credentials = [ uri.user || opts[:username], uri.password || opts[:password]].compact.presence
```

Reply:
readability suffers with this AI suggestion IMO. Might be because I'm not used to that construct but I vote to leave it as is.

Reply:
OK then :+1: